### DISCUSSION TITLE
How ro run a raw command using runscript in BatchAdminCommand?

### DISCUSSION THREAD

**Discussion by budachst** [CrowdStrike/falconpy#981] [Q&A] [answered]
I want to run a dscl on macOS to change some user's password. I successfully did that using a single id as a target using a service class. I am trying to get that working using an uber class approach, but the command alsways gets rejected with this error

40006, 'message': 'Command is not valid'. I am constructing my BODY payload like this:

```
{
  "base_command": "runscript",
  "command_string": "-Raw=```dscl . -passwd /Users/someuser %s```" % (newPW),
  "batch_id": "BATCHID",
  "optional_hosts": [
    "HOST1",
    "HOST2"
  ],
  "persist_all": true
}
```

If I leave the -Raw= and the backticks out, the command gets executed and the result returned, but I need the -Raw option, since I want to run an OS specific command.

**Comment by jshcodes** [comment, marked as answer]
> I want to run a dscl on macOS to change some user's password. I successfully did that using a single id as a target using a service class. I am trying to get that working using an uber class approach, but the command alsways gets rejected with this error
> 
> 40006, 'message': 'Command is not valid'. I am constructing my BODY payload like this:
> 
> ```
> {
>   "base_command": "runscript",
>   "command_string": "-Raw=```dscl . -passwd /Users/someuser %s```" % (newPW),
>   "batch_id": "BATCHID",
>   "optional_hosts": [
>     "HOST1",
>     "HOST2"
>   ],
>   "persist_all": true
> }
> ```
> 
> If I leave the -Raw= and the backticks out, the command gets executed and the result returned, but I need the -Raw option, since I want to run an OS specific command.

Does using a `command_string` set to 
```python
"command_string": "runscript -Raw=```dscl . -passwd /Users/someuser %s```" % (newPW)
```
work?

**Reply by budachst** [reply]
Oh wow… that was counter-intuitive, but that did the trick! Thank you!
